### ISSUE TITLE
Improve Integration with Google Drive (Chromebook)

### ISSUE THREAD

**Issue by carofe82** [keeweb/keeweb#289]
I use KeeWeb on my Chromebook and it is working awesome. Currently, when I add a new entry, I have to download the file and replace the version in Google Drive with my downloaded version manually. I have the file in a specific location to share it with other people and devices.
I would like to be able to select the specific folder where the KeePass file is, or when I go to Google Drive I would like to be able to do right click on the kdbx file and have KeeWeb available. The current Google Drive sync feature doesn't allow you to specify the folder the file is at.

I think you would need to make KeeWeb as a connected app for Google Drive.
I think this link may be related with how to implement it: https://developers.google.com/drive/v3/web/install.

Thank you!

**Comment by antelle** [comment]
Hello!
> download the file and replace the version in Google Drive 

Hm, why doesn't built-in Google Drive sync work for you? I mean, Click More &rarr; Google Drive &rarr; ...

> right click on the kdbx file and have KeeWeb available

We have to add 'Open with' command, you're right.

**Comment by carofe82** [comment]
> Hm, why doesn't built-in Google Drive sync work for you? I mean, Click More → Google Drive → ...

There are two ways of syncing with Google Drive:
* Save To: This option doesn't allow you to specify a folder inside Google Drive
* Open / Google Drive: This option doesn't show you where the file is, but just a list of found files without telling you where they are from.

**Comment by antelle** [comment]
Yes, you cannot select the folder but once selected, you will know which file it is.
Folder selection is not implemented currently, yes.

**Comment by carofe82** [comment]
I also see you have scheduled chromeos support for 1.4. What's the better support that will go in that version?

**Comment by antelle** [comment]
Yes, we're planning to create a chrome app and add the app to Drive's 'open with...' menu (use 'install' method).

**Comment by carofe82** [comment]
Excellent! Thank you.

The problem with opening a file from Google Drive is that it shows me three files when I only have 1. The Google Drive Search shows three files also. I think it is because of the way Google Drive works when sharing folders (keeping copies for collaborators when you replace files and so).

**Comment by antelle** [comment]
This is strange, most probably it shows files in trash? Do you have a trashed file with the same name?

**Comment by carofe82** [comment]
Trash is empty. It is just like a weird thing that Google Drive does. If you share a folder, and you replace a file inside that folder, Google "keeps" a copy of the original file to the collaborators, whatever that means. I deleted the other copies of the files directly from the search result and now KeeWeb shows me only one option when opening from Google Drive, great :).

**Comment by antelle** [comment]
That's strange, there should be a flag to determine that. I'll check this, thanks.
